Implements the  Acceptance Strategy

tiagomsleao opened the pull request:
The function of the percentage is still linear instead of exponential
Closes #1

tiagomsleao committed:
Implements the  Acceptance Strategy
The function of the percentage is still linear instead of exponential
Closes #1

pmisteliac reviewed (dismissed):
The strategy looks good, but I have some comments on the code formatting. Java variable and function names are in camel case by convention. So it would be reservationValue instead of reservation_value.
I don't mean to be overly criticizing but code quality accounts for 20% of our grade so I think we should pay attention to detail here because this is easy to achieve.

RensHam commented on AitGroup25Assignment/src/ai2018/group25/Group25_AS.java:
```diff
@@ -34,21 +37,48 @@ public void init(NegotiationSession negotiationSession, OfferingStrategy offerin
+		
+		// Initialize variable
+		double myLastBidUtil = -1.0;
+		double right_limit = -1.0;
```
As already said try to only use camelcase.

RensHam commented on AitGroup25Assignment/src/ai2018/group25/Group25_AS.java:
```diff
@@ -34,21 +37,48 @@ public void init(NegotiationSession negotiationSession, OfferingStrategy offerin
+		double right_limit = -1.0;
+		
+		// Get my last bid and the bid I am planning on doing next
+		if (negotiationSession.getOwnBidHistory().getLastBidDetails()!=null) { // Cover the first case
```
Documentation in code shouldn't be used to much. And please : 
```if (negotiationSession.getOwnBidHistory().getLastBidDetails() != null) {```

RensHam commented on AitGroup25Assignment/src/ai2018/group25/Group25_AS.java:
```diff
@@ -34,21 +37,48 @@ public void init(NegotiationSession negotiationSession, OfferingStrategy offerin
+		}
+		
+		// Get, according to time, the percentage of the interval not to accept
+		double percentage_interval_rejected = what_reject(negotiationSession.getTime()); // DO THIS FUNCTION
```
Remove comment //DO THIS FUNCTION

RensHam commented on AitGroup25Assignment/src/ai2018/group25/Group25_AS.java:
```diff
@@ -70,7 +102,16 @@ public String getName() {
-
+	
+	private double what_reject(double normalized_time) {
+		if ( normalized_time <= starts_falling ) { 
```
```if (val == val2) {```

tiagomsleao replied:
whats this?

RensHam commented on AitGroup25Assignment/src/ai2018/group25/Group25_AS.java:
```diff
@@ -70,7 +102,16 @@ public String getName() {
+		if ( normalized_time <= starts_falling ) { 
+			return 1.0;
+		} else {
+			return ( -1/(1-starts_falling) )*normalized_time + ( 1/(1-starts_falling) ); // Still linear, transform into exponential
```
It is more usefull to write javadoc above the function what it should do. Then this comment wouldn't be needed.

RensHam commented on AitGroup25Assignment/src/ai2018/group25/Group25_AS.java:
```diff
@@ -34,21 +37,48 @@ public void init(NegotiationSession negotiationSession, OfferingStrategy offerin
 	@Override
 	public Actions determineAcceptability() {
-		//TODO implement strategy here
+		// Accepting Strategy Implementation
```
Remove useless comments.

tiagomsleao replied:
it is not useless...

tiagomsleao committed:
Fixing some details on code formatting
Closes #1

tiagomsleao committed:
Changes to AS.java file and creation of important files
The Group25_AS.java has some minor changes, created the ConstBid.java file that has a constant bidding strategy just to test the AS and created a .zip file with the structure of what we will actually submit on
BrightSpace
Closes #1

RensHam commented on AitGroup25Assignment/src/ai2018/group25/ConstBid.java:
```diff
@@ -0,0 +1,170 @@
+	/** Minimum target utility */
+	private double Pmin;
+	/** Concession factor */
+	private double e;
```
I would go for
`private double concessionFactor` but that is upto you

tiagomsleao replied:
I justed used what was done and commented it all as you should have seen...

RensHam commented on AitGroup25Assignment/src/ai2018/group25/ConstBid.java:
```diff
@@ -0,0 +1,170 @@
+	 * From [1]:
+	 * 
+	 * A wide range of time dependent functions can be defined by varying the
+	 * way in which f(t) is computed. However, functions must ensure that 0 <=
```
:)
